rfcs/0011-releases.md

```diff
@@ -0,0 +1,73 @@
+
+## 2. Goals
+
+1. Define release numbering convention.
+2. Define release process steps.
+
+## 3. Method
+
+### 3.1. Release Numbers
+
+Each composable node release involves a release of (at-least) the following components,
+
+1. The Composable Node - main node executable
+
+    Node follows [semver](https://semver.org/) versioning with an additional `v` prefix. Eg: `v2.3.1`. 
+
+2. Runtime wasm - Dali, Picasso and Composable at the time of this writing.
+
+    In order to allow clear identification with the native runtime version of each node release the runtime version for each runtime is an integer derived by concatenating all numbers in the node version and adding to 10000. Eg: `v2.5.1` -> `251` -> 251 + 10000 -> (`dali-10251`, `picasso-10251`, `composable-10251`). The 10000 addition is to prevent collision of version numbers of this scheme with earlier versions before the scheme was introduced.
+
+3. Subsquid - Data archival and query system for networks.
+4. Frontends - There are two FE components in existence at the time of this writing.
+5. ComposableJs - This is the library to interact with composable parachains using typescript/JS.
+
+For 3, 4, 5 have the following characteristics which require them to be versioned in a different scheme,
+- They all depend on the node rpc and runtime (and types) directly hence requiring updates whenever the node RPC/event interfaces change.
+- They all can make breaking changes from the point of view of external dependants such as ComposableJS which could have major versions/breaking interface changes without a direct relationship to Node/runtime changes.
+
+Because of these characteristics following specific versioning scheme is proposed be used,
+
+`<component name>-<node v.major.minor version>-<component specific versioning>`
+
+Eg: 
+
+For Picasso FE : `fe-v2.4-picasso-abcd`
+
+For Subsquid : `subsquid-v2.5-1.0.1`
+
+etc.
+
+#### 3.1.1. Expected Typical Release Artefact List 
+
+```
+- Composable Node v2.5.3
+-- Runtimes
+--- dali-10253
+--- picasso-10253
+--- composable-10253
+-- FE
+--- fe-v2.5-picasso-abcd
+--- fe-v2.5-pablo-xyz
+-- ComposableJS
+--- composablejs-v2.5-mnop
+-- Subsquid
+--- subsquid-v2.5-111
+```
+
+### 3.2. Release Process
+
+Typical Composable releases involve multiple rounds of QA and external audits/testing that may cause multiple patch(rc) versions to be released based on feedback/issues. This means that a release branch may have a longer maintenance life cycle independent of the main branch where most of the bleeding edge development happens. In order to execute this expected workflow, following release process steps are proposed.
+
+As the work starts for a `major.minor` (eg: 2.5) release,
+
+1. Create a branch `release-v2.5`.
+2. in order to make/deploy (in staging) a release create a tag `release-v2.5.0` which should trigger a workflow.
```

Comment:
So you create a branch with release-xyz format, and then additionally tag that branch with a tag that is the same to the name? Shall we just omit 2?

Reply:
Branch format is different from tag. Branch is just major.minor version.

Reply:
So we're planning to skip minor version during QA? So we start of attempting to release v2.5, and end up with v2.5.6?

Reply:
Might it be better then to reduce it to a single number, like postgres does? Since the Major/Minor seem to not carry much information?

Reply:
Albula being release 1, but fixes to albula being 1.1, 1.14 etc?

Reply:
Sorry my wording was not correct before, the major version is always 2 above for v2.5, 5 being the minor. 

> Albula being release 1, but fixes to albula being 1.1, 1.14 etc?

I don't have a particular issue with this other than it breaking the previous scheme we were following for the node.

Reply:
We would then have that each component has an indepedent semver, compatible with the language standards of that ecosystem, and a monotonic increasing counter for the full release.

Reply:
If we are happy to break the previous scheme, which I am, then it could be nice to have this encapsulating version `Albula` with its subcomponents versioned independently.  @vimukthi-git I suppose this would mean a change or two right?

Reply:
Had a call with @KaiserKarel on this, so the discussion points were,
1. What are the benefits of having a `major.minor.patch` version, when we could probably manage with just `major.patch`. 
2. Then another idea was that we could use `major=spec_version` and then node patches without runtime change could be `spec_version.patch`.

After discussion I feel 2) may complicate the branching proposed here with each release having a branch. So I propose going with `major.minor` where `major=branch name eg: 5` and `minor=spec_version eg: 2400` for the node main version. Where 
1. Each node only patch release would require a new branch, with `spec_version` remaining the same. 
2. Each runtime only patch requires keeping updates in the same branch as before.

Reply:
Discussed with @andyjsbell and we will do following
```
Do major.minor where major=branch name eg: 5 and minor=spec_version eg: 2400 for the node main version
```